### ISSUE TITLE
Inclusion & exclusion predicates

### ISSUE THREAD

**Issue by fran-worley** [dry-rb/dry-logic#8]
The current inclusion and exclusion predicates take a list and check that the input is included in that list.

I have just run into a use case where I want to turn this on its head and instead check that the input includes a value.

```ruby
#current predicate:
predicate(:inclusion?) do |list, input|
  list.include?(input)
end

#what I need...
predicate(:some_new_predicate?) do |value, input|
  input.include?(value)
end
```

Does this seem like something worth adding to the library? 

In terms of names I guess we either go for `:includes?` and `:excludes?` the idea being that we are checking if an input includes or excludes something. However they might be a little similar to the current predicates...

**Comment by solnic** [comment]
Yes it is, you're not the first person who asked for it. IIRC we thought about naming it `contains?` but I'd be ok with `includes?` and `excludes?` although there's a risk people will be confused with `inclusion?` vs `includes?` - but I gotta say I never ever liked `inclusion?`, we picked it up because people are familiar with it. Blah

**Comment by fran-worley** [comment]
I thought of contains but what do you do for the opposite? I wasn't happy with anything I came up with (does_not_contain, missing, absent).

You could do:

Input.includes?(value) 
Input.included_in?(list) (instead of inclusion)

Input.excludes?(value)
Input.excluded_from?(list) (instead of exclusion)

**Comment by fran-worley** [comment]
Or (might be a bit much magic) you could determine what to call `.include?` on based on the type of argument supplied similar to how the Size predicate works.

**Comment by timriley** [comment]
I like your suggestion, @fran-worley:

```ruby
input.includes?(value) 
input.included_in?(list) #instead of inclusion

input.excludes?(value)
input.excluded_from?(list) #instead of exclusion
```

It looks a lot clearer to me, and it seems @solnic isn't attached to the current naming. Now might also be a good time to get it in, given we're changing the basic DSL anyway.

Would you be up for making this changing (and deprecating the other names) along with https://github.com/dry-rb/dry-logic/pull/9?

**Comment by fran-worley** [comment]
Yup will do. I'll also PR dry-v and the docs to update .

**Comment by fran-worley** [comment]
closed via #9
